### ISSUE TITLE
Traefik does not pickup changes to Consul Catalog in v1.4.0-rc1

### ISSUE THREAD

**Issue by davidgenn** [containous/traefik#2072]
### Do you want to request a *feature* or report a *bug*?

bug

### What did you do?

Had traefik successfully running in front of Consul/Nomad under v1.3.5.
Whenever a Nomad deploy updated the Consul catalog, this was reflected in Traefik.

Upgraded to v1.4.0-rc1 of Traefik.
The initial Consul catalog was loaded successfully.
Subsequent changes to Consul eg new service, updated service were not picked up by Traefik.

### Output of `traefik version`: (_What version of Traefik are you using?_)
v1.4.0-rc1

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
debug=true
traefikLogsFile = "/var/log/traefik/traefik.log"
accessLogsFile = "/var/log/traefik/access.log"
logLevel = "DEBUG"


[consulCatalog]

endpoint = "127.0.0.1:8500"

domain = "service.consul"

frontEndRule = "Host:{{ServiceName}}.{{Domain}},PathPrefixStrip:/{{ServiceName}}"

prefix = "traefik"
```

**Comment by davidgenn** [comment]
Log output:

```
time="2017-09-06T09:37:15Z" level=debug msg="Round trip: http://10.0.3.18:24699, code: 200, duration: 3.882703ms" 
time="2017-09-06T09:37:15Z" level=debug msg="Round trip: http://10.0.5.185:28900, code: 200, duration: 5.150997ms" 
time="2017-09-06T09:37:15Z" level=debug msg="Round trip: http://10.0.3.18:24153, code: 200, duration: 3.319474ms" 
time="2017-09-06T09:37:20Z" level=debug msg="Round trip: http://10.0.5.185:23347, code: 200, duration: 5.538646ms" 
time="2017-09-06T09:37:20Z" level=debug msg="Round trip: http://10.0.4.118:20081, code: 200, duration: 8.332484ms" 
time="2017-09-06T09:37:20Z" level=debug msg="Round trip: http://10.0.3.18:21750, code: 200, duration: 3.334634ms" 
time="2017-09-06T09:37:20Z" level=debug msg="Round trip: http://10.0.5.185:20896, code: 200, duration: 10.879997ms" 
time="2017-09-06T09:37:23Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 3.700693ms" 
time="2017-09-06T09:37:24Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 5.798822ms" 
time="2017-09-06T09:37:25Z" level=debug msg="Round trip: http://10.0.3.18:23234, code: 200, duration: 3.550983ms" 
time="2017-09-06T09:37:25Z" level=debug msg="Round trip: http://10.0.3.18:24699, code: 200, duration: 3.577458ms" 
time="2017-09-06T09:37:25Z" level=debug msg="Round trip: http://10.0.5.185:28900, code: 200, duration: 5.246275ms" 
time="2017-09-06T09:37:25Z" level=debug msg="Round trip: http://10.0.3.18:24153, code: 200, duration: 4.22121ms" 
time="2017-09-06T09:37:28Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 3.594038ms" 
time="2017-09-06T09:37:30Z" level=debug msg="Round trip: http://10.0.5.185:23347, code: 200, duration: 5.149131ms" 
time="2017-09-06T09:37:30Z" level=debug msg="Round trip: http://10.0.4.118:20081, code: 200, duration: 4.946183ms" 
time="2017-09-06T09:37:30Z" level=debug msg="Round trip: http://10.0.3.18:21750, code: 200, duration: 3.29993ms" 
time="2017-09-06T09:37:30Z" level=debug msg="Round trip: http://10.0.5.185:20896, code: 200, duration: 5.018212ms" 
time="2017-09-06T09:37:33Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 5.89856ms" 
time="2017-09-06T09:37:35Z" level=debug msg="Round trip: http://10.0.3.18:23234, code: 200, duration: 3.322102ms" 
time="2017-09-06T09:37:35Z" level=debug msg="Round trip: http://10.0.3.18:24699, code: 200, duration: 3.222127ms" 
time="2017-09-06T09:37:35Z" level=debug msg="Round trip: http://10.0.5.185:28900, code: 200, duration: 4.952389ms" 
time="2017-09-06T09:37:35Z" level=debug msg="Round trip: http://10.0.3.18:24153, code: 200, duration: 3.623628ms" 
time="2017-09-06T09:37:37Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 3.655476ms" 
time="2017-09-06T09:37:39Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 4.266689ms" 
time="2017-09-06T09:37:44Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 3.725204ms" 
time="2017-09-06T09:37:54Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 4.43126ms" 
time="2017-09-06T09:37:59Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 3.640577ms" 
time="2017-09-06T09:38:10Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 6.236057ms" 
time="2017-09-06T09:38:15Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 3.787467ms" 
time="2017-09-06T09:38:20Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 5.722215ms" 
time="2017-09-06T09:38:21Z" level=warning msg="Error forwarding to http://10.0.5.185:25158, err: dial tcp 10.0.5.185:25158: getsockopt: connection refused" 
time="2017-09-06T09:38:30Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 4.801746ms" 
time="2017-09-06T09:38:41Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 5.437244ms" 
time="2017-09-06T09:38:42Z" level=warning msg="Error forwarding to http://10.0.3.18:27864, err: dial tcp 10.0.3.18:27864: getsockopt: connection refused" 
time="2017-09-06T09:38:43Z" level=warning msg="Error forwarding to http://10.0.5.185:25158, err: dial tcp 10.0.5.185:25158: getsockopt: connection refused" 
time="2017-09-06T09:38:52Z" level=warning msg="Error forwarding to http://10.0.3.18:27864, err: dial tcp 10.0.3.18:27864: getsockopt: connection refused" 
time="2017-09-06T09:39:01Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 2.778986ms" 
time="2017-09-06T09:39:06Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 4.576876ms" 
time="2017-09-06T09:39:21Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 3.995806ms" 
time="2017-09-06T09:39:21Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 5.765099ms" 
time="2017-09-06T09:39:26Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 2.771504ms" 
time="2017-09-06T09:39:41Z" level=debug msg="Round trip: http://10.0.5.185:24585, code: 200, duration: 5.313999ms" 
time="2017-09-06T09:39:46Z" level=debug msg="Round trip: http://10.0.3.18:29532, code: 200, duration: 2.885707ms" 
```

The connection refused occurred after a Nomad deploy which destroyed 2 allocations for an existing service and moved them to a new node.

**Comment by ldez** [comment]
Closed by #2089.

**Comment by traefiker** [comment]
Closed by #2148.

**Comment by davidgenn** [comment]
👍
